rand_distr/tests/cdf.rs

```diff
@@ -159,6 +158,377 @@ fn normal() {
+    ];
+
+    for (seed, (lambda, k)) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::Weibull::new(lambda, k).unwrap();
+        test_continuous(seed as u64, dist, |x| cdf(x, lambda, k));
+    }
+}
+
+#[test]
+fn gumbel() {
+    fn cdf(x: f64, mu: f64, beta: f64) -> f64 {
+        (-(-(x - mu) / beta).exp()).exp()
+    }
+
+    let parameters = [
+        (0.0, 1.0),
+        (1.0, 2.0),
+        (-1.0, 0.5),
+        (10.0, 0.1),
+        (100.0, 0.0001),
+    ];
+
+    for (seed, (mu, beta)) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::Gumbel::new(mu, beta).unwrap();
+        test_continuous(seed as u64, dist, |x| cdf(x, mu, beta));
+    }
+}
+
+#[test]
+fn frechet() {
+    fn cdf(x: f64, alpha: f64, s: f64, m: f64) -> f64 {
+        if x < m {
+            return 0.0;
+        }
+
+        (-((x - m) / s).powf(-alpha)).exp()
+    }
+
+    let parameters = [
+        (0.5, 2.0, 1.0),
+        (1.0, 1.0, 1.0),
+        (10.0, 0.1, 1.0),
+        (100.0, 0.0001, 1.0),
+        (0.9999, 2.0, 1.0),
+    ];
+
+    for (seed, (alpha, s, m)) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::Frechet::new(m, s, alpha).unwrap();
+        test_continuous(seed as u64, dist, |x| cdf(x, alpha, s, m));
+    }
+}
+
+#[test]
+fn zeta() {
+    fn cdf(k: i64, s: f64) -> f64 {
+        if k < 1 {
+            return 0.0;
+        }
+
+        gen_harmonic(k as u64, s) / zeta_func(s)
+    }
+
+    let parameters = [2.0, 3.7, 5.0, 100.0];
+
+    for (seed, s) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::Zeta::new(s).unwrap();
+        test_discrete(seed as u64, dist, |k| cdf(k, s));
+    }
+}
+
+#[test]
+fn zipf() {
+    fn cdf(k: i64, n: u64, s: f64) -> f64 {
+        if k < 1 {
+            return 0.0;
+        }
+        if k > n as i64 {
+            return 1.0;
+        }
+        gen_harmonic(k as u64, s) / gen_harmonic(n, s)
+    }
+
+    let parameters = [(1000, 1.0), (500, 2.0), (1000, 0.5)];
+
+    for (seed, (n, x)) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::Zipf::new(n, x).unwrap();
+        test_discrete(seed as u64, dist, |k| cdf(k, n, x));
+    }
+}
+
+#[test]
+fn gamma() {
+    fn cdf(x: f64, shape: f64, scale: f64) -> f64 {
+        if x < 0.0 {
+            return 0.0;
+        }
+
+        (x / scale).inc_gamma(shape)
+    }
+
+    let parameters = [
+        (0.5, 2.0),
+        (1.0, 1.0),
+        (10.0, 0.1),
+        (100.0, 0.0001),
+        (0.9999, 2.0),
+    ];
+
+    for (seed, (shape, scale)) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::Gamma::new(shape, scale).unwrap();
+        test_continuous(seed as u64, dist, |x| cdf(x, shape, scale));
+    }
+}
+
+#[test]
+fn chi_squared() {
+    fn cdf(x: f64, k: f64) -> f64 {
+        if x < 0.0 {
+            return 0.0;
+        }
+
+        (x / 2.0).inc_gamma(k / 2.0)
+    }
+
+    let parameters = [
+        // 0.01, // Fail case
```

Comment:
This case should also be fine to drop (for now). The distribution has very high Kurtosis, so there will a significant amount of mass in the tails.